crates/rattler_package_streaming/src/reqwest/tokio.rs

```diff
@@ -131,12 +140,44 @@ pub async fn extract_conda(
     reporter: Option<Arc<dyn DownloadReporter>>,
 ) -> Result<ExtractResult, ExtractError> {
     // The `response` is used to stream in the package data
-    let reader = get_reader(url.clone(), client, expected_sha256, reporter.clone()).await?;
-    let result = crate::tokio::async_read::extract_conda(reader, destination).await?;
-    if let Some(reporter) = &reporter {
-        reporter.on_download_complete();
+    let reader = get_reader(
+        url.clone(),
+        client.clone(),
+        expected_sha256,
+        reporter.clone(),
+    )
+    .await?;
+    match crate::tokio::async_read::extract_conda(
+        reader,
+        destination,
+        crate::read::extract_conda_via_streaming,
+    )
+    .await
+    {
+        Ok(result) => {
+            if let Some(reporter) = &reporter {
+                reporter.on_download_complete();
+            }
+            Ok(result)
+        }
+        // https://github.com/conda-incubator/rattler/issues/794
+        Err(ExtractError::ZipError(zip_error))
+            if (zip_error
+                .to_string()
+                .contains(DATA_DESCRIPTOR_ERROR_MESSAGE)) =>
+        {
+            tracing::debug!("Failed to stream decompress conda package from '{}' due to the presence of zip data descriptors. Falling back to non streaming decompression;", url);
```

Comment:
A bit split between using `debug` or `warn` here. I'm tempted to use WARN since this is a non-advisable state to be in.